net/quagga/src/etc/inc/plugins.inc.d/quagga.inc

```diff
@@ -26,6 +26,43 @@
     POSSIBILITY OF SUCH DAMAGE.
 */
 
+function ospfd_enabled()
+{
+    $model = new \OPNsense\Quagga\OSPF();
+    if ((string)$model->enabled == '1') {
+        return true;
+    }
+
+    return false;
+}
+
+function quagga_firewall($fw)
+{
+    if (ospfd_enabled()) {
+        $ospf = new \OPNsense\Quagga\OSPF();
+        foreach ($ospf->networks->network->__items as $network) {
+            if ((string)$network->enabled == '1') {
+                $fw->registerFilterRule(
+                    1, /* priority */
+                    array(
+                        'ipprotocol'     => 'inet',
+                        'protocol'       => 'ospf',
+                        'statetype'      => 'keep',
+                        'label'          => 'Pass OPSF (autogenerated)',
+                        'from'           => $network->ipaddr . '/' . $network->netmask,
+                        'to'             => '224.0.0.0/4',
+                        'direction'      => 'in',
+                        'type'           => 'pass',
+                        'disablereplyto' => 1,
```

Comment:
you sure? this could break multi-wan

Reply:
OSPF runs between neighbors - it will not survive another hop.

Reply:
"disablereplyto" means reply where you want, not necessarily through this link? isn't that what you don't want?

Reply:
ok, if the hop is not the router it will fail...

Reply:
It is multicast traffic. I don't want to force a gateway. The routes in the same network will talk to each other in this multicast group. OSPF is not routed.